File: limitless/backend/app/schemas/message.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MessageCreate(BaseModel):
    content: str
    event_time: datetime | None = None


class MessageResponse(BaseModel):
    id: UUID
    tenant_id: UUID
    user_id: UUID
    content: str
    event_time: datetime
    ingested_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: limitless/frontend/src/types/message.ts
export interface MessageCreate {
  content: string;
  event_time?: string;
}

export interface MessageResponse {
  id: string;
  tenant_id: string;
  user_id: string;
  content: string;
  event_time: string;
  ingested_at: string;
}

export interface MessageListResponse {
  total: number;
  page: number;
  limit: number;
  items: MessageResponse[];
}
